### PR TITLE
V3ify docker docs

### PR DESCRIPTION
**Pull request by harrietgrace**
Did some updates for the Containerized Builds with Docker doc:
- Moved the env vars info to new sections about using Docker Compose/Docker without plugins
- Update the `docker-compose.yml` example
- Update the "Configuring the build step" section with plugin examples and descriptions
- Update the "Docker without docker compose" section with plugin examples and descriptions
- Add links to the plugins github readmes
- Add info about `docker system prune` to the cleanup section

### CODE REVIEW

File: pages/builds/docker_containerized_builds.md.erb [hunk 1]
```diff
@@ -75,24 +67,32 @@ Once you've got everything working, you can configure your build pipeline and ru
...
-This will tell the agent to run each build step like so:
+At the start of your `pipeline.yml`, add a step to build an image with the docker plugin. In this example, we use the `docker-compose` plugin to prebuild the `app` image and upload it to a registry:
 
-```bash
-docker-compose -f docker-compose.yml run app <command>
+```
+steps:
+  - name: "Docker Build"
+    plugins:
+      docker-compose#v1.3.2:
```
Comment (lox) [inline pages/builds/docker_containerized_builds.md.erb]: Ugh this version is going to be such a pain to keep up to date.

File: pages/builds/docker_containerized_builds.md.erb [hunk 2]
```diff
@@ -101,11 +101,56 @@ To allow buildkite-agent to use the Docker client you’ll need to ensure its us
...
+
+There are many configuration options available for the Docker plugin. For the complete list, see the readme for the [Docker Buildkite Plugin on Github](https://github.com/buildkite-plugins/docker-buildkite-plugin).
+
+## Creating Your Own Docker Infrastructure
+
+If your team has significant Docker experience you might find it worthwhile invoking your own runner scripts rather than using the simpler built-in Docker support.
+
+To do this see the [agent hooks](/docs/agent/v2/hooks) and [parallelizing builds](parallelizing-builds) documentation.
+
+## Using Docker Compose without plugins
```
Comment (lox) [inline pages/builds/docker_containerized_builds.md.erb]: Maybe we should make it clear that this is deprecated? "Using docker compose via the deprecated built-in support" or something?

File: pages/builds/docker_containerized_builds.md.erb [hunk 3]
```diff
@@ -75,24 +67,32 @@ Once you've got everything working, you can configure your build pipeline and ru
...
 
-Firstly we need to add the following environment variable to our build, either on an individual build step or as a pipeline-level environment variable:
-
-```bash
-BUILDKITE_DOCKER_COMPOSE_CONTAINER=app
-```
+To run your steps in a container, we recommend using [plugins](/docs/agent/v3/plugins). If you want to use docker without plugins, see [Using Docker without plugins](/docs/builds/docker-containerized-builds#using-docker-without-plugins)
 
-This will tell the agent to run each build step like so:
+At the start of your `pipeline.yml`, add a step to build an image with the docker plugin. In this example, we use the `docker-compose` plugin to prebuild the `app` image and upload it to a registry:
```
[the remark this answers is not in the document]
Comment (harrietgrace) [reply pages/builds/docker_containerized_builds.md.erb]: Added in!

File: pages/builds/docker_containerized_builds.md.erb (hunk 1 above)
Comment (harrietgrace) [reply pages/builds/docker_containerized_builds.md.erb]: Should we bother keeping it up to date? I could include a note about ensuring that you're pinning the plugin to the most recent version or something?

File: pages/builds/docker_containerized_builds.md.erb (hunk 2 above)
Comment (harrietgrace) [reply pages/builds/docker_containerized_builds.md.erb]: It would be good to have a 'deprecated' warning block we could add in to any piece of docs... I'll try one out here and we can re use it elsewhere!

File: pages/builds/docker_containerized_builds.md.erb (hunk 1 above)
Comment (lox) [reply pages/builds/docker_containerized_builds.md.erb]: Yeah, that might be good. Let's go with `v2.0.0` so it's up to date anyway!

**Comment by lox** [comment]
This is looking awesome @harrietgrace! Nice one!

File: pages/builds/docker_containerized_builds.md.erb (hunk 1 above)
Comment (harrietgrace) [reply pages/builds/docker_containerized_builds.md.erb]: Added this in:
image

File: pages/builds/docker_containerized_builds.md.erb (hunk 2 above)
Comment (harrietgrace) [reply pages/builds/docker_containerized_builds.md.erb]: It's a bit wordy, but I wasn't sure how else to give all the info:

image

I've put another one in the Docker without plugins bit too
